### PR TITLE
docs: edits to Four kinds of data

### PR DESCRIPTION
**Pull request by peknis**
Language and formatting fixes.

Signed-off-by: Pekka Niskanen <pekka.niskanen@nordicsemi.no>

### CODE REVIEW

**Review by umapraseeda — APPROVED** [review]

File: docs/guides/FourKindsOfData.rst [hunk 1]
```diff
@@ -1,21 +1,21 @@
 .. _four-kinds-data:
 
-Four kinds of data
+Four types of data
```
Comment (coderbyheart) [inline docs/guides/FourKindsOfData.rst]: Title is renamed, but not filename and therefore URL.

File: docs/guides/FourKindsOfData.rst [hunk 2]
```diff
@@ -25,124 +25,125 @@ Following are the four types of data that is used in the communication of the de
...
-It might not be favorable for a device to be connected all the time since wireless communication is relatively expensive, consumes a lot of energy thereby increasing the power consumption in the device.
+An important criterion for the robustness of any IoT product is the graceful handling of situations where the device is not connected to the internet.
+It is not feasible for a device to be connected all the time since wireless communication is relatively expensive, consumes a lot of energy thereby increasing the power consumption in the device.
 
-The firmware is optimized for ultra-low power consumption, and the design aims in reducing the modem uptime and keep the modem in the off state as long as possible.
-This is achieved by making the device smart and allowing it to decide on sending the data based on the situation.
+The firmware is optimized for ultra-low power consumption, and the design aims in reducing the modem uptime and keeping the modem in the off state as long as possible.
+A smart device decides on sending the data based on the situation.
```
Comment (coderbyheart) [inline docs/guides/FourKindsOfData.rst]: This changes the meaning, now the just exists a smart device, before we made a device smart.

File: docs/guides/FourKindsOfData.rst [hunk 3]
```diff
@@ -25,124 +25,125 @@ Following are the four types of data that is used in the communication of the de
...
 Upgrading a fleet of devices with a new firmware involves orchestrating the roll-out and monitoring the faults.
-All these challenges mandate the ability to *configure the device*, which allows to tweak the behavior of the device until the inflection point is reached with respect to battery life and data granularity.
+All these challenges mandate the ability to configure the device, which allows to tweak the behavior of the device until the inflection point is reached with respect to battery life and data granularity.
 
 An interesting configuration option is the sensitivity of the motion sensor.
-The action that is considered as a *movement* varies from one tracked subject to another.
+An action that is considered as a movement varies from one tracked subject to another.
 A device has various timeout settings such as the waiting time to acquire a GPS fix, or the waiting time between updates that are sent when the device is in motion.
-The timeout settings have an important influence on power consumption and data consumption.
+The timeout settings have a remarkable influence on power and data consumption.
```
Comment (coderbyheart) [inline docs/guides/FourKindsOfData.rst]: This changes the meaning.
Remarkable is not a synonym for important.
If important can't be used, how about significant?

File: docs/guides/FourKindsOfData.rst [hunk 4]
```diff
@@ -25,124 +25,125 @@ Following are the four types of data that is used in the communication of the de
...
 
 There are many scenarios where the cellular connection might not be available or might be unreliable, but the reading sensors work.
-Robust ultra-mobile IoT products therefore must incorporate such conditions into the normal mode of operation.
+Robust ultra-mobile IoT products must incorporate such conditions into the normal mode of operation.
 The absence of a cellular connection must be treated as a temporary condition, which will eventually resolve and until then normal mode of operation must continue.
 This means that the devices must continue to measure and store these measurements in a ring buffer or employ other strategies to decide on the data to be discarded once the memory limit is reached.
 
-Once the device can establish a connection successfully, it will publish the past data in batch (after publishing its most recent measurements).
-Since the device memory is limited, a strategy must be created to discard old messages.
-A simple approach is to use a ring buffer that stores the latest messages and will discard the oldest message once the size limit is reached.
```
Comment (coderbyheart) [inline docs/guides/FourKindsOfData.rst:123]: Please don't delete important information.

**Review by coderbyheart — CHANGES_REQUESTED** [review]
Please review the changes I highlighted and avoid changing the meaning. Please also do not remove entire segments.

File: docs/guides/FourKindsOfData.rst (hunk 1 above)
Comment (peknis) [reply docs/guides/FourKindsOfData.rst]: Does it matter?

File: docs/guides/FourKindsOfData.rst (hunk 3 above)
Comment (peknis) [reply docs/guides/FourKindsOfData.rst]: remarkable and significant are pretty much synonyms.

File: docs/guides/FourKindsOfData.rst (hunk 4 above)
Comment (peknis) [reply docs/guides/FourKindsOfData.rst:123]: The information I left out is pretty much redundant. It is repeating the same story you just told above.

File: docs/guides/FourKindsOfData.rst (hunk 3 above)
Comment (coderbyheart) [reply docs/guides/FourKindsOfData.rst]: So, can we use significant?

File: docs/guides/FourKindsOfData.rst (hunk 4 above)
Comment (coderbyheart) [reply docs/guides/FourKindsOfData.rst:123]: I see, given that context I understand the reason for the deletion.

File: docs/guides/FourKindsOfData.rst (hunk 1 above)
Comment (coderbyheart) [reply docs/guides/FourKindsOfData.rst]: Yes, I keep all page URLs aligned with the titles of the page, so that this create no confusion for the reader.

Comment (peknis) [reply docs/guides/FourKindsOfData.rst]: Okay. The easiest solution is to change it back to what it originally was. Fixing it now.

**Pull request merged**